Empire - The Last Battle/Assets/Unity/Scripts/MonoBehaviours/OverworldManager.cs

```diff
@@ -185,7 +187,50 @@ void _StormshapersPlayer_OnCardRemoved(CardData card, int index)
 
 	}
 
+	void _PointsSystem_CurrencyChanged(int val)
+	{
+		_OverworldUI._ArmouryUI_OnCurrencyChanged(val, _GameStateHolder._ActivePlayer);
+	}
+
+	//Listen to OnShowToggled here because we need to call endTurn and get active player
+	public void _OverworldUI_ArmouryUI_OnShowToggled(bool toggledOn)
```

Comment:
Not sure that i understand this? this isn't a listener and is just called by the function below?

I would have throught a better way would be to call _OverworldUI.ShowArmoury() down below, and handle all this in the UI manager. You could listen to showToggled (seperate OnHide and OnShow would be nicer imo ) and end the turn when you heard hide.

But again like i mentioned before, if we're hiding the UI on pause, then it will end the turn in that case too if you pause while on the armoury ui (unless i've missed something in the changes )